tests/test_multistage.py

```diff
@@ -0,0 +1,276 @@
+    u_multi_stage = TimeFunction(name='u_multi_stage', grid=grid, space_order=2, time_order=1, dtype=float64)
+
+    # Source definition
+    src_spatial = Function(name="src_spat", grid=grid, space_order=2, dtype=float64)
+    src_spatial.data[1, 1] = 1
+    src_temporal = (1 - 2 * (t*dt - 1)**2)
+
+    # PDE
+    eq_rhs = (Derivative(u_multi_stage, (x, 2), fd_order=2) +
+              Derivative(u_multi_stage, (y, 2), fd_order=2) +
+              src_spatial * src_temporal)
+
+    # Time integration scheme
+    pde = [resolve_method(time_int)(u_multi_stage, eq_rhs)]
+    op = Operator(pde, subs=grid.spacing_map)
+    op(dt=0.01, time=1)
+
+
+@pytest.mark.parametrize('time_int', ['RK44', 'RK32', 'RK97'])
+def test_multistage_methods_convergence(time_int):
+    extent = (1000, 1000)
+    shape = (201, 201)
+    origin = (0, 0)
+
+    # Grid setup
+    grid = Grid(origin=origin, extent=extent, shape=shape, dtype=float64)
+    x, y = grid.dimensions
+    dt = grid.stepping_dim.spacing
+    t = grid.time_dim
+    dx = extent[0] / (shape[0] - 1)
+
+    # Medium velocity model
+    vel = Function(name="vel", grid=grid, space_order=2, dtype=float64)
+    vel.data[:] = 1.0
+    vel.data[150:, :] = 1.3
+
+    # Source definition
+    src_spatial = Function(name="src_spat", grid=grid, space_order=2, dtype=float64)
+    src_spatial.data[100, 100] = 1/dx**2
+    f0 = 0.01
+    src_temporal = (1-2*(pi*f0*(t*dt-1/f0))**2)*exp(-(pi*f0*(t*dt-1/f0))**2)
+
+    # Time axis
+    t0, tn = 0.0, 500.0
+    dt0 = max(vel.data)/dx**2
+    nt = int((tn-t0)/dt0)
+    dt0 = tn/nt
+
+    # Time integrator solution
+    # Define wavefield unknowns: u (displacement) and v (velocity)
+    fun_labels = ['u', 'v']
+    U_multi_stage = [TimeFunction(name=name+'_multi_stage', grid=grid, space_order=2, time_order=1, dtype=float64) for name in fun_labels]
+
+    # PDE (2D acoustic)
+    eq_rhs = [U_multi_stage[1], (Derivative(U_multi_stage[0], (x, 2), fd_order=2) +
+                                 Derivative(U_multi_stage[0], (y, 2), fd_order=2) +
+                                 src_spatial * src_temporal) * vel**2]
+
+    # Time integration scheme
+    pdes = [resolve_method(time_int)(U_multi_stage[i], eq_rhs[i]) for i in range(len(fun_labels))]
+    op = Operator(pdes, subs=grid.spacing_map)
+    op(dt=dt0, time=nt)
+
+    # Devito's default solution
+    U = [TimeFunction(name=name, grid=grid, space_order=2, time_order=1, dtype=float64) for name in fun_labels]
+    # PDE (2D acoustic)
+    eq_rhs = [U[1], (Derivative(U[0], (x, 2), fd_order=2) + Derivative(U[0], (y, 2), fd_order=2) +
+                     src_spatial * src_temporal) * vel**2]
+
+    # Time integration scheme
+    pdes = [Eq(U[i].forward, solve(Eq(U[i].dt-eq_rhs[i]), U[i].forward)) for i in range(len(fun_labels))]
+    op = Operator(pdes, subs=grid.spacing_map)
+    op(dt=dt0, time=nt)
+    assert max(abs(U[0].data[0,:]-U_multi_stage[0].data[0,:]))<10**-5, "the method is not converging to the solution"
```

Comment:
A better check would be to compare the L2 norms with `np.isclose`. See other tests in the codebase for examples